include/aspect/utilities.h

```diff
@@ -47,6 +47,27 @@ namespace aspect
     using namespace dealii;
     using namespace dealii::Utilities;
 
+
+    /**
+     * Split the set of DoFs (typically locally owned or relevant) in @p whole_set into blocks
+     * given by the @p dofs_per_block structure.
+     */
+    inline void split_by_block (std::vector<IndexSet> &partitioned,
+                                const std::vector<types::global_dof_index> &dofs_per_block,
+                                const IndexSet &whole_set)
```

Comment:
Move the output argument to the end.

The function is expensive. I don't think there's a point in making it `inline`.